operations/observability/mixins/workspace/rules/components/workspaces/alerts.libsonnet

```diff
@@ -73,6 +73,23 @@
               gitpod_workspace_regular_not_active_percentage > 0.15 AND sum(gitpod_ws_manager_workspace_activity_total) > 100
             |||,
           },
+          {
+            alert: 'GitpodWorkspacesNotStarting',
+            labels: {
+              severity: 'critical',
+            },
+            'for': '10m',
```

Comment:
Hi @princerachit does it make sense to leave this at 10m, when the expression is inspecting rate for 5m? :thinking: Seems odd to me, but, maybe I am missing something.

Reply:
I can change this but it does not make any difference, afterall we are checking if the value is zero

Reply:
Can you share if this would have triggered in the last 30 or 60 days with support detail? 

My understanding is `for` controls how long an expression must be true before the alert fires. And the underlying expression is looking for a window of time where the sum of starts is 0.

I would expect this alert to fire after 20 minutes. For 10 minutes, the following must be true:
1. regular not active has been > than zero
2. no workspaces have started in the last 10 minutes

I think it would have triggered if we were using 5m and 5m, triggered [here](https://grafana.gitpod.io/?orgId=1&var-datasource=VictoriaMetrics&var-cluster=us38xl2&from=1649419599314&to=1649422342364).

![image](https://user-images.githubusercontent.com/1272076/162476702-7fccc8ed-739e-43a9-a59f-852a3dfe4736.png)

Reply:
@princerachit marked as unresolved so I can get your [feedback](https://github.com/gitpod-io/gitpod/pull/9190#discussion_r846264519).

Reply:
> I would expect this alert to fire after 20 minutes.

That is 100% correct, 

When using `rate(metric[10m])`, the value will be calculated using the latest 10 minutes of data. This means that the value cannot increase for the last 10 minutes for this value ever becomes 0. Adding a `for` to the alert will just increase the time for the alert to be triggered.

For the alert to be triggered in 10 minutes you can:
* rate[5m] + for: 5m
* rate[10m]

Reply:
[Some extra info on rates](https://promlabs.com/blog/2021/01/29/how-exactly-does-promql-calculate-rates) :)